label/service.go

```diff
@@ -0,0 +1,283 @@
+package label
+
+import (
+	"context"
+	"strings"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kv"
+)
+
+type Service struct {
+	store *Store
+	kvSvc *kv.Service
+}
+
+func NewService(st *Store, kvSvc *kv.Service) influxdb.LabelService {
+	return &Service{
+		store: st,
+		kvSvc: kvSvc, // todo (al) this can be removed once URMs are removed from the Label service
+	}
+}
+
+// CreateLabel creates a new label.
+func (s *Service) CreateLabel(ctx context.Context, l *influxdb.Label) error {
+	if err := l.Validate(); err != nil {
+		return &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Err:  err,
+		}
+	}
+
+	l.Name = strings.TrimSpace(l.Name)
+
+	err := s.store.Update(ctx, func(tx kv.Tx) error {
+		if err := uniqueLabelName(ctx, tx, l); err != nil {
+			return err
+		}
+
+		if err := s.store.CreateLabel(ctx, tx, l); err != nil {
+			return err
+		}
+
+		if err := s.kvSvc.CreateUserResourceMappingForOrg(ctx, tx, l.OrgID, l.ID, influxdb.LabelsResourceType); err != nil {
```

Comment:
I just tried to implement this and ran into the problem I had when I initially tried to copy this function over, which is that it calls functions on the kv service in order to do the update (and those each in turn call other functions on the service). I could be missing something but I don't know if it's possible to access this code without including a kv service in there somewhere, even if it were on this other interface

Reply:
(just kidding, Lyon explained to me what you meant! I was trying to avoid passing the kv service at all which wasn't working but now I'm just using this to wrap it)

Reply:
I did, on Lyon's suggestion, add a wrapping interface that just includes the `CreateUserResourceMappingForOrg` function to remove the direct dependence on kv. However, with this I'll still need to export the function in kv and pass a kv service where the label service gets initialized.